### PR TITLE
Add AbortController

### PR DESCRIPTION
**Pull request by caub**

### CODE REVIEW

File: polyfills/AbortController/config.json [hunk 1]
```diff
@@ -20,8 +20,7 @@
     "install": {
         "module": "abortcontroller-polyfill",
         "paths": [
-            "dist",
-            "umd-polyfill.js"
+            "dist/umd-polyfill.js"
```
Comment (caub) [inline polyfills/AbortController/config.json]: :+1:

**Comment by caub** [comment]
@JakeChampion There's https://github.com/mysticatea/abort-controller that's maybe more spec compliant (I participated to mo/abortcontroller-polyfill that's why I knew about it, and thought of it first)

**Comment by JakeChampion** [comment]
@caub I have already submitted the PR, I linked to it in my commebt above --> https://github.com/mo/abortcontroller-polyfill/pull/27

**Comment by weyert** [comment]
Any news on this? Looks like the issue in polyfill project has been resolved. What's needed to get this merged?

**Comment by JakeChampion** [comment]
@weyert It needs to pass all the tests but currently is failing quite a few of them. If you have the time, I'd really appreciate help looking into how to make the polyfill pass the tests --> https://circleci.com/gh/Financial-Times/polyfill-library/2159?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by caub** [comment]
I'll make a PR for changing the .babelrc there https://github.com/mo/abortcontroller-polyfill/blob/master/.babelrc it's probably not good enough for IE10

https://github.com/mysticatea/abort-controller

**Comment by caub** [comment]
Oh, I think I know why https://github.com/mo/abortcontroller-polyfill/pull/27 didn't work on IE10

you used directly `Object.defineProperty` but it's quite broken on IE10, we need to use es-shims for that

I'll update https://github.com/mo/abortcontroller-polyfill/pull/34 with it

**Comment by JakeChampion** [comment]
@caub I have push your forked branch up to our remote, which will trigger CircleCI to run. Thanks for working on this :-)

**Comment by JakeChampion** [comment]
@caub Here are the test results from CircleCI, sorry the report is so large. 

```
SUMMARY:
✔ 704 tests completed
✖ 176 tests failed

FAILED TESTS:
  AbortSignal
    basic tests
      ✖ abort during fetch
        IE 10.0.0 (Windows 7 0.0.0)
      Invalid operand to 'in': Object expected (lib/index.js:1317)

      ✖ abort during fetch when Request has signal
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      done() invoked with non-Error: {"name":"fail"}
      polyfills/AbortController/tests.js:62:21

      ✖ abort before fetch started
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      done() invoked with non-Error: {"name":"fail"}
      polyfills/AbortController/tests.js:83:21

      ✖ fetch without aborting
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1293:9)
         at Anonymous function (lib/index.js:1486:11)
         at f (lib/index.js:505:292)
         at rn (lib/index.js:505:2831)
         at abortableFetch (lib/index.js:1485:9)
         at Anonymous function (polyfills/AbortController/tests.js:133:13)

      ✖ fetch without signal set
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
      done() invoked with non-Error: {"name":"fail"}
      @polyfills/AbortController/tests.js:144:17
      

      ✖ event listener fires "abort" event
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1293:9)
         at Anonymous function (polyfills/AbortController/tests.js:163:13)

      ✖ signal.aborted is true after abort
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1293:9)
         at Anonymous function (polyfills/AbortController/tests.js:174:13)

      ✖ event listener doesn't fire "abort" event after removeEventListener
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1293:9)
         at Anonymous function (polyfills/AbortController/tests.js:195:13)

      ✖ toString() output
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      [object AbortController] deepStrictEqual [object AbortSignal]

      ✖ AbortSignal constructor
        Chrome 68.0.3440 (Android 4.4.2)
      TypeError: Illegal constructor
          at Context.<anonymous> (polyfills/AbortController/tests.js:24:26)
```

**Comment by caub** [comment]
Ok, let's try this now ^ now :)

**Comment by JakeChampion** [comment]
@caub It looks like the polyfill is now using `const`, which is a feature that lots of the browsers won't have. Those browsers will see it as a syntax error. Are you able to make the polyfill use only `var` instead of `const` and `let`?

**Comment by caub** [comment]
woa, it's not better with abort-controller

✔ 426 tests completed
✖ 454 tests failed

I'll commit with abortcontroller-polyfill with the correct module dep this time

**Comment by JakeChampion** [comment]
Most of the failing tests look to be because of this error `Invalid operand to 'in': Object expected`

**Comment by mo** [comment]
@JakeChampion did CI not finish for f7e8d31 or what was the results?

**Comment by JakeChampion** [comment]
@caub & @mo CI is now running, sorry for the delay, I was on holiday and away from my computer

**Comment by JakeChampion** [comment]
Here are the tests which failed and in which browsers --
```
FAILED TESTS:
  AbortSignal
    basic tests
      ✖ AbortSignal constructor
        Chrome 68.0.3440 (Android 4.4.2)
      TypeError: Illegal constructor
          at Context.<anonymous> (polyfills/AbortController/tests.js:24:26)

      ✖ abort during fetch
        IE 10.0.0 (Windows 7 0.0.0)
      Invalid operand to 'in': Object expected (lib/index.js:2005)

      ✖ abort during fetch when Request has signal
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      done() invoked with non-Error: {"name":"fail"}
      polyfills/AbortController/tests.js:62:21

      ✖ abort before fetch started
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      done() invoked with non-Error: {"name":"fail"}
      polyfills/AbortController/tests.js:83:21

      ✖ fetch without aborting
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1978:9)
         at Anonymous function (lib/index.js:2263:11)
         at f (lib/index.js:894:292)
         at rn (lib/index.js:894:2831)
         at abortableFetch (lib/index.js:2262:9)
         at Anonymous function (polyfills/AbortController/tests.js:133:13)

      ✖ event listener fires "abort" event
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1978:9)
         at Anonymous function (polyfills/AbortController/tests.js:163:13)

      ✖ signal.aborted is true after abort
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1978:9)
         at Anonymous function (polyfills/AbortController/tests.js:174:13)

      ✖ event listener doesn't fire "abort" event after removeEventListener
        IE 10.0.0 (Windows 7 0.0.0)
      TypeError: Invalid operand to 'in': Object expected
         at addEventListener (lib/index.js:1978:9)
         at Anonymous function (polyfills/AbortController/tests.js:195:13)

      ✖ toString() output
        Chrome 36.0.1985 (Windows XP 0.0.0)
        Chrome 35.0.1916 (Windows XP 0.0.0)
        Chrome 38.0.2125 (Windows XP 0.0.0)
        Chrome 37.0.2062 (Windows XP 0.0.0)
        Chrome 68.0.3440 (Android 4.4.2)
        Chrome 39.0.2171 (Windows XP 0.0.0)
        Chrome 40.0.2214 (Windows XP 0.0.0)
        Chrome 41.0.2272 (Windows XP 0.0.0)
        Chrome 43.0.2357 (Windows XP 0.0.0)
        Chrome 44.0.2403 (Windows XP 0.0.0)
        Chrome 45.0.2454 (Windows XP 0.0.0)
        Chrome 46.0.2490 (Windows XP 0.0.0)
        Chrome 47.0.2526 (Windows XP 0.0.0)
        Chrome 48.0.2564 (Windows XP 0.0.0)
        Chrome 49.0.2623 (Windows XP 0.0.0)
        Chrome 50.0.2661 (Windows 7 0.0.0)
        Chrome 51.0.2704 (Windows 7 0.0.0)
        Chrome 52.0.2743 (Windows 7 0.0.0)
        Chrome 54.0.2840 (Windows 7 0.0.0)
        Chrome 53.0.2785 (Windows 7 0.0.0)
        Chrome 55.0.2883 (Windows 7 0.0.0)
        Chrome 57.0.2987 (Windows 7 0.0.0)
        Chrome 56.0.2924 (Windows 7 0.0.0)
        Chrome 58.0.3029 (Windows 7 0.0.0)
        Chrome 59.0.3071 (Windows 7 0.0.0)
        Chrome 60.0.3112 (Windows 7 0.0.0)
        Chrome 61.0.3163 (Windows 7 0.0.0)
        Chrome 63.0.3239 (Windows 7 0.0.0)
        Chrome 64.0.3282 (Windows 7 0.0.0)
        Chrome 62.0.3202 (Windows 7 0.0.0)
        Chrome 65.0.3325 (Windows 7 0.0.0)
        Edge 15.15063.0 (Windows 10 0.0.0)
        Firefox 38.0.0 (Windows XP 0.0.0)
        Firefox 40.0.0 (Windows XP 0.0.0)
        Firefox 41.0.0 (Windows XP 0.0.0)
        Firefox 42.0.0 (Windows XP 0.0.0)
        Firefox 43.0.0 (Windows XP 0.0.0)
        Firefox 44.0.0 (Windows XP 0.0.0)
        Firefox 45.0.0 (Windows XP 0.0.0)
        Firefox 46.0.0 (Windows XP 0.0.0)
        Firefox 47.0.0 (Windows XP 0.0.0)
        Firefox 48.0.0 (Windows 7 0.0.0)
        Firefox 49.0.0 (Windows 7 0.0.0)
        Firefox 50.0.0 (Windows 7 0.0.0)
        Firefox 51.0.0 (Windows 7 0.0.0)
        Firefox 52.0.0 (Windows 7 0.0.0)
        Firefox 53.0.0 (Windows 7 0.0.0)
        Firefox 54.0.0 (Windows 7 0.0.0)
        Firefox 55.0.0 (Windows 7 0.0.0)
        IE 10.0.0 (Windows 7 0.0.0)
        Firefox 56.0.0 (Windows 7 0.0.0)
        IE 11.0.0 (Windows 7 0.0.0)
        Safari 10.1.2 (Mac OS X 10.12.6)
        Safari 11.1.2 (Mac OS X 10.13.6)
        Safari 9.1.3 (Mac OS X 10.11.6)
      [object AbortController] deepStrictEqual [object AbortSignal]
```

**Comment by mo** [comment]
In abortcontroller-polyfill 71bbd1f I had quite a lot of changes to the testcases, among these changes I completely removed the "AbortSignal constructor" testcase because it was incorrect (it didn't pass when the native abortcontroller was used). The testcases on abortcontroller-polyfill master are in much better shape and they are verified to pass both against the native abortcontroller and with the polyfill installed.

**Comment by JakeChampion** [comment]
@mo it looks like there is a bug in the polyfill when it is running on IE10 whereby the listeners property is not set and the dispatchEvent method does not handle the case where this.listeners is not an object, after looking into the code being executed, I can't see the Emitter constructor being called. I came to that conclusion by adding a breakpoint to the line which adds the listeners property.

image

bs_win7_IE_10 0
bs_win7_IE_10 0 (1)

**Comment by mo** [comment]
@JakeChampion Note that when I run abortcontroller-polyfill 1.2.9 via a standalone testapp in IE8/9/10/11 all of them just work. However, I was able to reproduce the error you saw when I was running via your karma testrunner. It seems that the "super()" call in AbortSignal is not transpiled correctly by babel which causes the EventEmitter constructor to never run. Googling for "babel ie10 parent constructor" has lots of info on such issues.

After trying out a few fixes I decided on explicitly calling the parent constructor for cases when it failed to run, so I have released abortcontroller-polyfill 1.3.0 which has a fix for your issue.

When I run the tests with abortcontroller-polyfill upgraded to 1.3.0 I still get one test fail in IE10 related to Object.prototype.toString.call(new AbortController()) !== '[object AbortController]'

When I debugged this a bit I noticed that the "if (typeof Symbol !== 'undefined' && Symbol.toStringTag) {" branch inside abortcontroller-polyfill wasn't running.

I then added "Symbol.toStringTag" to the "dependencies" section in config.json in polyfill-library and that causes the above if statement to run properly (which is good), but it unfortunately was not sufficient to fix the Object.prototype.toString.call(new AbortController()) !== '[object AbortController]' test failure.

I think it would be good if you could A) upgrade to 1.3.0, and B) add "Symbol.toStringTag" to "dependencies" in config.json

**Comment by SimplySayHi** [comment]
Hi guys,
any update on this?

Thanks :)

**Comment by JakeChampion** [comment]
Hi @SimplySayHi, there is no update on this. It currently is failing some tests and I don't have the time right now to look into it. If you are willing and able, please feel free to look into the failing tests, once every test succeeds I will merge the pull-request :-)

**Comment by mo** [comment]
Great work @JakeChampion

**Comment by JakeChampion** [comment]
It's live now on https://cdn.polyfill.io :-)

**Pull request merged**